pkg/minikube/audit/audit.go

```diff
@@ -51,14 +53,34 @@ func args() string {
 }
 
 // Log details about the executed command.
-func Log(startTime time.Time) {
+func LogCommandStart() (string, err) {
 	if len(os.Args) < 2 || !shouldLog() {
-		return
+		return "", fmt.Errorf("This command should not be logged and len(os.Args) : %v, should be less than 2", len(os.Args))
 	}
-	r := newRow(os.Args[1], args(), userName(), version.GetVersion(), startTime, time.Now())
+	id := uuid.New().String()
+	r := newRow(os.Args[1], args(), userName(), version.GetVersion(), time.Now(), id)
 	if err := appendToLog(r); err != nil {
-		klog.Warning(err)
+		return "", klog.Warning(err)
 	}
+	return r.id, nil
+}
+
+func LogCommandEnd(id string) err{
+	if currentLogFile == nil {
+		if err := setLogFile(); err != nil {
+			return fmt.Errorf("failed to set the log file: %v", err)
+		}
+	}
+	auditLogs, err := logsToRows(currentLogFile)
```

Comment:
@spowelljr How can we convert `currentLogFile` from `*os.File` type to `[]string` type. We can use `ioutil.ReadFile()` function for converting it to []byte and then string but how can we convert it to []string type so that we can convert to []row through `logsToRows()` function.

Reply:
You can use https://pkg.go.dev/encoding/json#Unmarshal to convert `[]byte` to a struct (row).